ledger/participant-state/kvutils/src/main/scala/com/daml/ledger/validator/preexecution/LogAppenderPreExecutingCommitStrategy.scala

```diff
@@ -17,13 +23,35 @@ import scala.concurrent.{ExecutionContext, Future}
 
 final class LogAppenderPreExecutingCommitStrategy(
     keySerializationStrategy: StateKeySerializationStrategy,
-) extends PreExecutingCommitStrategy[RawKeyValuePairsWithLogEntry] {
+) extends PreExecutingCommitStrategy[
+      DamlStateKey,
+      (Option[DamlStateValue], Fingerprint),
+      ReadSet,
+      RawKeyValuePairsWithLogEntry,
+    ] {
   private val stateSerializationStrategy = new StateSerializationStrategy(keySerializationStrategy)
 
+  override def generateReadSet(
+      fetchedInputs: Map[DamlStateKey, (Option[DamlStateValue], Fingerprint)],
```

Comment:
I think this was a bit more readable with a type alias.

Reply:
Agreed, but it's going away entirely soon.